### DISCUSSION TITLE
What are the steps that ESPixelStick Flash Tool performs?

### DISCUSSION THREAD

**Discussion by Terrevue** [forkineye/ESPixelStick#803] [Q&A] [answered]
No matter what I do, I can't get the Flash Tool to work on MacOS Sequoia.  I can run the python serial files just fine though and connect to the console:

python bin/upload.py --chip esp32 --baud 460800 --port /dev/cu.usbserial-1120 erase_flash
esptool.py v2.8
Serial port /dev/cu.usbserial-1120
Connecting.....
Chip is ESP32D0WDQ5 (revision 3)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 08:a6:f7:b1:14:d4
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Erasing flash (this may take a while)...
Chip erase completed successfully in 3.2s
Hard resetting via RTS pin...

No idea why the flash tool doesn't work but if I could get the steps that it performs, I can just run those directly without using the tool at all.

**Comment by Terrevue** [comment, marked as answer]
Nevermind.  I found it.  Here they are in the event someone else needs this:

python bin/upload.py --chip esp32 --baud 115200 --port /dev/cu.usbserial-1120 --before default_reset --after hard_reset write_flash -z 0x1000 firmware/esp32/esp32_devkitc-bootloader.bin 0x8000 firmware/esp32/esp32_devkitc-partitions.bin 0xe000 firmware/esp32/boot_app0.bin 0x10000 firmware/esp32/esp32_devkitc-app.bin 0x3B0000 firmware/filesystem.bin

Obviously change the port, firmware, etc to those specific to your build.

**Comment by MartinMueller2003** [comment]
Or use the espswebflashtool

**Reply by Terrevue** [reply]
I usually do but the latest build isn't appearing there yet.

**Reply by dutch-anykey** [reply]
It doesn't appear with a number.

https://github.com/forkineye/ESPixelStick/issues/802#issuecomment-2525052204

**Comment by MartinMueller2003** [comment]
https://espixelstickwebflasher.from-ct.com:5000/

**Comment by MartinMueller2003** [comment]
The absolute latest build will be up as "experimental". It has all of the latest changes and is in the testing phase.
